src/crush/CrushWrapper.cc

```diff
@@ -3128,10 +3128,28 @@ class CrushTreeFormattingDumper : public CrushTreeDumper::FormattingDumper {
     : Parent(crush, wsnames, show_shadow) {}
 
   void dump(Formatter *f) {
+    f->open_object_section("crush_tree");
```

Comment:
@OshynSong first, we normally don't print the object section for the object being printed in its own `dump()` method, we do so at the caller site. in this case, it's `OSDMonitor::preprocess_command()`. i'd suggest you add the missing `f->open_object_section("crush_tree");` and `f->close_section();` out there.

secondly, i assume you are printing the stray OSDs. as we don't have "stray" bucket in Ceph. if that's the case, probably you should just dump "leaves",  and exclude buckets in "stray" section. as `name_map` also include buckets IIUC. and i'd also recommend you put the "stray" change in a separate commit in this PR.

Reply:
@tchaikov  The two problems
 have been solved , see the new commit in https://github.com/ceph/ceph/pull/23319/commits/fa1056cfc32ce3bf932d7c71f281c09078c36999.